tests/core/test_expectation_suite.py

```diff
@@ -409,24 +414,93 @@ def test_cloud_context_update_suite_adds_ids(
             uuid_to_test = expectation.id
             assert isinstance(UUID(uuid_to_test), UUID)
 
+    @pytest.mark.unit
+    def test_suite_add_expectation_doesnt_allow_adding_an_expectation_with_id(
+        self, expectation
+    ):
+        suite = ExpectationSuite("test-suite")
+        provided_id = "6b3f003d-d97b-4649-ba23-3f4e30986297"
+        expectation.id = provided_id
+
+        with pytest.raises(
+            RuntimeError,
+            match="Cannot add Expectation because it already belongs to an ExpectationSuite.",
+        ):
+            suite.add(expectation)
+
+    @pytest.mark.cloud
+    def test_cloud_expectation_can_be_saved_after_added(
```

Comment:
This test and the following are identical except for the context fixture. To avoid repetition you could parameterize it, or, I think easier, make a helper function:

```
def _test_can_save_after_add(self, context, expectation):
    # test code you have here

@pytest.mark.cloud
def test_cloud_expectation_can_be_saved_after_added(
    self, empty_cloud_context_fluent, expectation
):
    self._test_can_save_after_add(empty_cloud_context_fluent, expectation)

@pytest.mark.filesystem
def test_cloud_expectation_can_be_saved_after_added(
    self, empty_data_context, expectation
):
    self._test_can_save_after_add(empty_cloud_context_fluent, expectation)

```

Reply:
https://greatexpectationslabs.slack.com/archives/C04ST5TSR71/p1703260461204929?thread_ts=1703259519.730379&cid=C04ST5TSR71

Reply:
thanks - I've updated across the board